### PR TITLE
Added support for double in batch norm

### PR DESCRIPTION
**Pull request by snnn**
**Description**:

Support double type for BatchNormalization operator

**Motivation and Context**
- Why is this change required? What problem does it solve?
To replace #2915 

- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by AlirezaAlizadeh** [comment]
Thanks for the fix!
